### PR TITLE
Optimize position of login popup for multi-monitor setups

### PR DESCRIPTION
**Pull request by PoornimaVKrishnan**
This PR is to optimize position of login popup for multi-monitor setups. It centre aligns the popup in chrome and firefox also opens the pop up in the right screen.

The solution for this is to calculate the left and top positions of the popup from the screen resolutions. The solution I referred to: https://stackoverflow.com/questions/4068373/center-a-popup-window-on-screen

### CODE REVIEW

File: src/providers/authn.tsx [hunk 1]
```diff
@@ -170,31 +170,29 @@ export default function AuthnProvider({ children }: AuthnProviderProps): JSX.Ele
...
 
-    // left should be in the middle of the screen
-    const popupLeft = (screen.availWidth - popupWidth) / 2;
-    // top should just have some small offset if there's available space
-    const popupTop = (topOffset + popupHeight) < screen.availHeight ? topOffset : 0;
+    const primaryScreen = window.screen;
 
+    // Calculate the center position based on the primary screen dimensions and popup size
+    const centerLeft = Math.max(0, (primaryScreen.availWidth - popupWidth) / 2 + primaryScreen.availLeft);
+    const centerTop = Math.max(0, (primaryScreen.availHeight - popupHeight) / 2 + primaryScreen.availTop);
```
Comment (RFSH) [inline src/providers/authn.tsx:180]: Let's leave this to what it was. We still want the popup to show up closer to the top of the page, so it's more visible.

**Review by RFSH — CHANGES_REQUESTED** [review]
The link that you added to PR is to the question not the answer. So please add the link to the answer.

Also `availLeft` and `availTop` are non-standard. So if we can avoid them, we should. What's the point of having those in the calculation?

You can see that in [here](https://developer.mozilla.org/en-US/docs/Web/API/Screen/availLeft) it's suggesting that we should avoid using this attribute if we can.

**Comment by PoornimaVKrishnan** [comment]
> The link that you added to PR is to the question not the answer. So please add the link to the answer.
> 
> Also `availLeft` and `availTop` are non-standard. So if we can avoid them, we should. What's the point of having those in the calculation?
> 
> You can see that in [here](https://developer.mozilla.org/en-US/docs/Web/API/Screen/availLeft) it's suggesting that we should avoid using this attribute if we can.

The solution I pointed out. It is what I referred. I googled on how else to get the screen left offset. And went through MDN docs and saw availLeft and availTop. We need to add those so that it aligns in the centre. It represents the left position of the available screen space, excluding areas occupied by other system elements.
I have added jsdoc that explains everything, And on the position from top have reverted changes as we need  popup to show up closer to the top of the page.
When going through all the solutions I realised that we need to add a left offset. I avoided using window.screenLeft or window.screenX to solely depend on screen calculations instead of window

File: src/providers/authn.tsx (hunk 1 above)
Comment (PoornimaVKrishnan) [reply src/providers/authn.tsx:180]: reverted

File: src/providers/authn.tsx [hunk 2]
```diff
@@ -170,31 +170,29 @@ export default function AuthnProvider({ children }: AuthnProviderProps): JSX.Ele
...
+    const popupHeight = Math.min(750, screen.availHeight);
 
-    // left should be in the middle of the screen
-    const popupLeft = (screen.availWidth - popupWidth) / 2;
-    // top should just have some small offset if there's available space
-    const popupTop = (topOffset + popupHeight) < screen.availHeight ? topOffset : 0;
+    const primaryScreen = window.screen;
 
+    // Calculate the center position based on the primary screen dimensions and popup size
+    const centerLeft = Math.max(0, (primaryScreen.availWidth - popupWidth) / 2 + primaryScreen.availLeft);
```
[the remark this answers is not in the document]
Comment (PoornimaVKrishnan) [reply src/providers/authn.tsx:179]: have added jsdoc

**Review by RFSH — CHANGES_REQUESTED** [review]
We want the popup to be displayed in the middle of the screen (horizontally). The position of the popup should not be dependent on the window's position or size. 

The previous implementation with `availLeft` was doing what we wanted. But the API is non-standard, and therefore we should avoid it.


Your new implementation is not doing what we want. In scenarios where the left of your browser is not stretched to the left, you can see that the popup is not centered anymore, and it's taking the position of your current window into account.

This is a simple task in terms of expected behavior. I don't understand why we have to go back and forth multiple times. Make sure you're testing your code instead of just pushing something and asking for feedback.

**Comment by RFSH** [comment]
Since @PoornimaVKrishnan couldn't find any solution that works consistently, we decided to just close this PR.

Refer to [this comment](https://github.com/informatics-isi-edu/chaise/issues/2189#issuecomment-1605178843) for more information.

**Pull request closed without merging**